### PR TITLE
fix: cartesian products in get_evaluations

### PR DESCRIPTION
**Pull request by RogerHYang**

### CODE REVIEW

**Review by anticorrelator — COMMENTED** [review]
do you think it's important to have a test for this one?

**Comment by RogerHYang** [comment]
> do you think it's important to have a test for this one?

we don't have a set up for server integration tests in general

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 1]
```diff
@@ -266,23 +258,14 @@ def _read_sql_document_evaluations_into_dataframe(
...
+        select(models.DocumentAnnotation, models.Span.span_id)
+        .join_from(models.DocumentAnnotation, models.Span)
+        .join_from(models.Span, models.Trace)
+        .join_from(models.Trace, models.Project)
+        .where(models.Project.name == project_name)
+        .where(models.DocumentAnnotation.annotator_kind == "LLM"),
         connectable,
-        index_col=["span_id", "document_position"],
-    )
+    ).set_index(["span_id", "document_position"])
```
Comment (axiomofjoy) [inline src/phoenix/server/api/routers/v1/evaluations.py:268]: Reason to prefer this to `index_col`?

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 2]
```diff
@@ -266,23 +258,14 @@ def _read_sql_document_evaluations_into_dataframe(
...
-                models.Project.name == project_name,
-                models.SpanAnnotation.annotator_kind == "LLM",
-            )
-        ),
+        select(models.DocumentAnnotation, models.Span.span_id)
+        .join_from(models.DocumentAnnotation, models.Span)
+        .join_from(models.Span, models.Trace)
+        .join_from(models.Trace, models.Project)
+        .where(models.Project.name == project_name)
+        .where(models.DocumentAnnotation.annotator_kind == "LLM"),
```
Comment (axiomofjoy) [inline src/phoenix/server/api/routers/v1/evaluations.py:266]: Good catch.

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 1 above)
Comment (RogerHYang) [reply src/phoenix/server/api/routers/v1/evaluations.py:268]: It didn't work. Don't know why

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 2 above)
Comment (axiomofjoy) [inline src/phoenix/server/api/routers/v1/evaluations.py:266]: So strange to me that sqlalchemy doesn't complain here.

**Comment by anticorrelator** [comment]
> > do you think it's important to have a test for this one?
> 
> we don't have a set up for server integration tests in general

I think we can seed data use SQLAlchemy and test the query at whatever level you find appropriate, something like here: https://github.com/Arize-ai/phoenix/blob/sql/tests/db/test_models.py, is this setup insufficient to catch this?

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py:268]: Weird

**Review by axiomofjoy — APPROVED** [review]
great catches, are we still doing cross joins elsewhere?

**Comment by RogerHYang** [comment]
> > > do you think it's important to have a test for this one?
> > 
> > 
> > we don't have a set up for server integration tests in general
> 
> I think we can seed data use SQLAlchemy and test the query at whatever level you find appropriate, something like here: https://github.com/Arize-ai/phoenix/blob/sql/tests/db/test_models.py, is this setup insufficient to catch this?

yea, that would work

**Pull request merged**